editor.planx.uk/src/@planx/components/shared/Schema/model.ts

```diff
@@ -0,0 +1,146 @@
+import { array, BaseSchema, object, ObjectSchema, string } from "yup";
+
+import { checklistValidationSchema } from "../../Checklist/model";
+import {
+  DateInput,
+  dateRangeSchema as dateValidationSchema,
+} from "../../DateInput/model";
+import { NumberInput, numberInputValidationSchema } from "../../NumberInput/model";
+import {
+  TextInput,
+  userDataSchema as textInputValidationSchema,
+} from "../../TextInput/model";
+import { Option } from "..";
+
+/**
+ * Simplified custom QuestionInput
+ * Existing model is too complex for our needs currently
+ * If adding more properties here, check if re-using existing model could be an option
+ */
+interface QuestionInput {
+  title: string;
+  description?: string;
+  options: Option[];
+}
+
+interface ChecklistInput {
+  title: string;
+  description?: string;
+  options: Option[];
+}
+
+/**
+ * As above, we need a simplified validation schema for QuestionsInputs
+ */
+const questionInputValidationSchema = (data: QuestionInput) =>
+  string()
+    .oneOf(data.options.map((option) => option.data.val || option.data.text))
+    .required("Select your answer before continuing");
+
+export type TextField = {
+  type: "text";
+  data: TextInput & { fn: string };
+};
+
+export type NumberField = {
+  type: "number";
+  data: NumberInput & { fn: string };
+};
+
+export type QuestionField = {
+  type: "question";
+  data: QuestionInput & { fn: string };
+};
+
+export type ChecklistField = {
+  type: "checklist";
+  required?: true;
+  data: ChecklistInput & { fn: string };
+};
+
+export type DateField = {
+  type: "date";
+  data: DateInput & { fn: string };
+};
+
+/**
+ * Represents the input types available in the List component
+ * Existing models are used to allow to us to re-use existing components, maintaining consistend UX/UI
+ */
+export type Field = TextField | NumberField | QuestionField | ChecklistField | DateField;
+
+/**
+ * Models the form displayed to the user
+ */
+export interface Schema {
+  type: string;
+  fields: Field[];
+  min: number;
+  max?: number;
+}
+
+export type SchemaResponse = Record<Field["data"]["fn"], string | string[]>;
+
+/**
+ * Output data from a form using the useSchema hook
+ */
+export type SchemaData = { schemaData: SchemaResponse[] };
```

Comment:
Why change these type names? `UserData` is used elsewhere in a few components and it means "the data generated by the user in this component". By renaming this we indicate that it's intrinsically linked to the data generated by just the schema (and may not represent the entire user output of a component). Thoughts welcome - let me know if this isn't a constructive change.

Reply:
Seems fine to me, I like that it flags that the component is a lot more generic.

Reply:
Agree `UserData` needed updating here, but are there any hesitations of `SchemaData` being a little too interchangable with payload jargon (eg how we commonly talk about _application output_)? 

Is `SchemaUserData` a happy medium, eg `UserData` suffix keeps mental link to _component output_ while still indicating it's about "Fields"? (Very nit picky, feel free to dismiss!)

Reply:
This works well for me! Fixed in [896e395](https://github.com/theopensystemslab/planx-new/pull/3526/commits/896e3954212d80ecd3dfc445ab9f8531e505e10f)